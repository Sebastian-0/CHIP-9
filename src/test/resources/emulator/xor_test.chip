# Test simple xors
ldi A, 0b01010101
ldi B, 0b01010010
ldi C, 0b10010101
ldi D, 0b11111111
ldi E, 0b00000000
ldi H, 0b11110000
ldi L, 0b00001111

xor B
xor C
xor D
xor E
xor H
xor L
xor A

sout A
sout B
sout C
sout D
sout E
sout H
sout L

ldi A, 0b01010101
ldx HL, 0x1010
ldi (HL), 0b10101010
xor (HL)

sout (HL)

# Test flags
ldi A, 0b01100110
ldi B, 0b01100110
ldi C, 0b11000001
ldi D, 0b01111001

# Zero flag
xor B
sout FLAGS

# Negative flag
xor C
sout FLAGS

# No flags
xor D
sout FLAGS

hcf