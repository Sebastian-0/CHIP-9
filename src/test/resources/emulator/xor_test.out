0b00000000
0b00000111
0b11000000
0b10101010
0b01010101
0b10100101
0b01011010
0b11111111

128
64
0